fighthealthinsurance/views.py

```diff
@@ -41,40 +51,43 @@ class ProVersionThankYouView(generic.TemplateView):
 
 class ProVersionView(generic.FormView):
     template_name = "professional.html"
-    form_class = InterestedProfessionalForm
+    form_class = core_forms.InterestedProfessionalForm
 
     def form_valid(self, form):
         form.save()
-        if (
+
+        if not (
             "clicked_for_paid" in form.cleaned_data
             and form.cleaned_data["clicked_for_paid"]
         ):
-            stripe.api_key = settings.STRIPE_API_SECRET_KEY
-            stripe.publishable_key = settings.STRIPE_API_PUBLISHABLE_KEY
-            product = stripe.Product.create(name="Pre-Signup")
-            product_price = stripe.Price.create(
-                unit_amount=1000, currency="usd", product=product["id"]
-            )
-            items = [
-                {
-                    "price": product_price["id"],
-                    "quantity": 1,
-                }
-            ]
-            checkout = stripe.checkout.Session.create(
-                line_items=items,
-                mode="payment",  # No subscriptions
-                success_url=self.request.build_absolute_uri(
-                    reverse("pro_version_thankyou")
-                ),
-                cancel_url=self.request.build_absolute_uri(
-                    reverse("pro_version_thankyou")
-                ),
-                customer_email=form.cleaned_data["email"],
-            )
-            checkout_url = checkout.url
-            return redirect(checkout_url)
-        return render(self.request, "professional_thankyou.html")
+            return render(self.request, "professional_thankyou.html")
+
+        stripe.api_key = settings.STRIPE_API_SECRET_KEY
+        stripe.publishable_key = settings.STRIPE_API_PUBLISHABLE_KEY
+
+        product = stripe.Product.create(name="Pre-Signup")
+        product_price = stripe.Price.create(
+            unit_amount=1000, currency="usd", product=product["id"]
+        )
```

Comment:
Maybe add a TODO here so that we re-use the product ID

Reply:
Not sure what you mean. Also, I was wondering if we should should be clearing the stripe keys or something here after this? Does stripe not give us an instantiated object that we can use for this instead of changing it globally?